server/rpc.go

```diff
@@ -297,6 +331,21 @@ func getOrchestrator(orch Orchestrator, req *net.OrchestratorRequest) (*net.Orch
 	return orchestratorInfo(orch, addr, orch.ServiceURI().String())
 }
 
+func endSession(node *core.LivepeerNode, orch Orchestrator, req *net.EndSessionRequest) (*net.EndSessionResponse, error) {
+	addr := ethcommon.BytesToAddress(req.Address)
```

Comment:
Instead of validating the B signature here perhaps we can just attach an AuthToken field as is done for the [SegData message](https://github.com/livepeer/go-livepeer/blob/e3f9286f80044dc1e55a4687eaeccb689a2ebf84/net/lp_rpc.proto#L197).

The AuthToken used in the EndSessionRequest should be the same AuthToken that the B was embedding in the SegData messages that was sending to O in segment requests. Then, when O receives the EndSessionRequest it can validate the Token field of the AuthToken message to make sure the request to end the session is authenticated using the AuthToken associated with that session - see [this example](https://github.com/livepeer/go-livepeer/blob/e3f9286f80044dc1e55a4687eaeccb689a2ebf84/server/segment_rpc.go#L399) for verifying the token. And we'd still be able to access the sessionId since it is included in the AuthToken message.